echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/PipelineConfigsPollingJob.groovy

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.scheduler.actions.pipeline
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.echo.model.Trigger
+import com.netflix.spinnaker.echo.pipelinetriggers.PipelineCache
+import groovy.util.logging.Slf4j
+import org.quartz.CronTrigger
+import org.quartz.DisallowConcurrentExecution
+import org.quartz.Job
+import org.quartz.JobExecutionContext
+import org.quartz.Scheduler
+import org.quartz.TriggerKey
+import org.quartz.impl.matchers.GroupMatcher
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Syncs triggers from pipelines with the triggers in the scheduler
+ */
+@Slf4j
+@DisallowConcurrentExecution
+class PipelineConfigsPollingJob implements Job {
+  private TimeZone timeZoneId
+  private Scheduler scheduler
+  private Registry registry
+  private PipelineCache pipelineCache
+
+  @SuppressWarnings("unused")
+  PipelineConfigsPollingJob() {
+    pipelineCache = (PipelineCache) SchedulerBeanDependencies.getBean(PipelineCache)
```

Comment:
i was wondering the same but forgot to investigate, thanks for reminding me will take a look! I suspect it's still needed since it's being constructed by `quartz` but will check

Reply:
Yeah, you're right. Can use the `SpringBeanJobFactory` annoyingly, it's more code. I guess it buys is better testability. I will take care of everything else first and will revisit this later